File: Low Level Design (LLD)/DesignPatterns/3. Structural/Strategy/Cpp/main.cpp
#include "include/PaymentService.h"
#include "include/CreditCardPayment.h"
#include "include/PayPalPayment.h"
#include "include/BitcoinPayment.h"

int main() {
    PaymentService service;

    CreditCardPayment creditCard("1234-5678-9012-3456");
    service.setPaymentStrategy(&creditCard);
    service.processPayment(250);

    PayPalPayment paypal("john@example.com");
    service.setPaymentStrategy(&paypal);
    service.processPayment(150);

    BitcoinPayment bitcoin("1A2B3C4D5E6F");
    service.setPaymentStrategy(&bitcoin);
    service.processPayment(100);

    return 0;
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Strategy/Java/Main.java
public class Main {
    public static void main(String[] args) {
        PaymentService service = new PaymentService();

        service.setPaymentStrategy(new CreditCardPayment("1234-5678-9012-3456"));
        service.processPayment(250);

        service.setPaymentStrategy(new PayPalPayment("john@example.com"));
        service.processPayment(150);

        service.setPaymentStrategy(new BitcoinPayment("1A2B3C4D5E6F"));
        service.processPayment(100);
    }
}
